### ISSUE TITLE
Verification fails when using arrays

### ISSUE THREAD

**Issue by Kasi-mir** [lpinca/shopify-token#8]
Hello,

Is there a reason why the verification would fail if there are arrays in the request. For example if you would want to verify the Action Drop Down (that contains multiple ids of products etc)?

Kasimir

**Comment by Kasi-mir** [comment]
Hello,

Issue resolution seems to be explained here: https://ecommerce.shopify.com/c/shopify-apis-and-technology/t/hmac-calculation-vs-ids-arrays-320575

Each array needs to be verified with key=[1, 2, 3]

Kasimir

**Comment by lpinca** [comment]
I think it fails because query strings with arrays are not handled properly: https://github.com/lpinca/shopify-token/blob/2a537c275070f5cdebb003fe7786b6145f9d2bc9/index.js#L108-L110.

Does Shopify explain how to treat arrays when dealing with arrays?

**Comment by Kasi-mir** [comment]
Yes, they don't seem to document it well, but been doing some testing and in for example ids it works with 

`ids["123341232", "12321332212", "1233212342"]`

It needs the whitespace aswell.

Kasimir

**Comment by lpinca** [comment]
Ok, but isn't it out of scope for this module? AFAIK the query string for the token does not include arrays.

**Comment by Kasi-mir** [comment]
I would not say so, as it validates the HMAC and this is something Shopify has decided to provide, its a query string that Shopify provides for multiple items.

**Comment by lpinca** [comment]
The module has been made for the access token and this is not needed for it. Something like this will probably work

```diff
diff --git a/index.js b/index.js
index f22ef51..284bcab 100644
--- a/index.js
+++ b/index.js
@@ -105,9 +105,15 @@ class ShopifyToken {
       .map((key) => {
         const value = query[key];
 
+        if (Array.isArray(value)) {
+
+        }
+
         return typeof value === 'string'
           ? `${encodeKey(key)}=${encodeValue(value)}`
-          : '';
+          : Array.isArray(value)
+            ? `${encodeKey(key)}=["${encodeValue(value.join('", "'))}"]`
+            : '';
       })
       .sort();
 
```

but I'm not very happy with it.

**Comment by lpinca** [comment]
There is no documentation so it's like playing a guessing game but anyway it should be fixed on master. Would you mind to confirm?

**Comment by Kasi-mir** [comment]
Perfect, working and confirmed.

Thanks

**Comment by lpinca** [comment]
Thanks, released 3.0.1.
I should also add support for the new online tokens but will do it in another release.
